### DISCUSSION TITLE
IPFS-powered static asset storage

### DISCUSSION THREAD

**Discussion by cofob** [deta/cloud-docs#255] [Ideas] [answered]
Hi. Now Deta Drive does not have the ability to give files by link directly, users have to do a proxy to give the file to the end client. This has several problems:

1. It's complicated, slow and unclear way. We already have storage, why drive traffic up and down when we can give it directly. This is an anti-pattern that doubles the amount of traffic.
2. The file can be large and the client's connection can be slow. The download can take more than 10 seconds, and then micro will timeout and the client won't get the file.

In the chat we talked about how the solution to this problem is being worked out, additionally I suggest to make a separate public static storage based on IPFS. Recently I tried to publish sites in this network, the results are impressive. In addition to decentralization, we get partial offloading of communication channels due to caching at both clients and gateways. The torrent for the modern internet. An example of a [site that runs entirely on an IPFS](https://site-in-ipfs.cofob.ru/) network on a public gateway.

I don't think you will take this seriously, as it is more complicated than just setting up s3, my job is to suggest it. Thanks in advance.

**Comment by abdelhai** [comment, marked as answer]
first, thanks for sharing this idea! as you predicted, although definitely interesting, we will just be offering a cdn for serving drive files sometime this year. this should solve both presented problems.

**Reply by cofob** [reply]
Additionally, I would like to ask that users be allowed to safely (without revealing critical keys) upload files larger than 5 mb directly to Drive. (From browser, for example)

**Reply by abdelhai** [reply]
good point. we need to figure this out. we're making large changes to our infra, so this might be a non-problem soon.
